### DISCUSSION TITLE
[nuxt-bridge] How can I access to props in <script setup> ?

### DISCUSSION THREAD

**Discussion by manudurgoni** [nuxt/nuxt#18198] [Questions]
Hello,

I just started a project with nuxt-bridge. And I'm trying to access to my props in `<script setup>`, without success.
In my example, I can access to `url` in my template. With Vue3, I use `<script setup="props">`, but it doesn't work with nuxt-bridge. Any ideas ?

Here an example : https://codesandbox.io/s/runtime-https-n0zes?file=/components/SliceCover.vue

```vue
// SliceCover.vue
<script>
import Slice from "./Slice.vue";
export default {
  extends: Slice,
};
</script>

<script setup>
import { onMounted } from "#app";

// Can I access to my "url" props here ?
</script>


<template>
  <div class="slice-cover">
    <h1>Cover Slice : {{ url }}</h1>
  </div>
</template>
```

```vue
// Slice.vue
// where I define my props
<script>
export default {
  props: {
    url: {
      type: String,
      required: true,
    },
  },
};
</script>
```

**Comment by jyotirmaybarman** [comment]
@manudurgoni 
You may try
```js
const props = defineProps({
  id: {
    type: String,
    required: true
  }
});
```
// and then access the prop by
``props.id``

**Comment by manudurgoni** [comment]
Oh thanks 👍 

But the idea is to extend the props from Slice.vue. Because I will have a lot of components `Slice{Type}.vue`.

```
const props = defineProps({
  id: {
    type: String,
    required: true
  }
});
```

It redefines the props in the extend component right ?

**Reply by jyotirmaybarman** [reply]
I couldn't find any better solution rather than this...!!

**Comment by joshkoppang** [comment]
Could you do something like:

```
// Put this in a shared location
interface SharedProps {
  prop1: string;
}

// Put this in the Slice{Type}.vue
interface ChildProps extends SharedProps {
  prop2: string;
}

const props = defineProps<LeaderboardProps>();
```
